extensions/amp-analytics/0.1/activity-impl.js

```diff
@@ -86,8 +92,10 @@ class ActivityHistory {
     }
 
     if (this.prevActivityEvent_.time < activityEvent.time) {
-      this.totalEngagedTime_ +=
-          findEngagedTimeBetween(this.prevActivityEvent_, activityEvent.time);
+      this.incrementalEngagedTime_ = findEngagedTimeBetween(
```

Comment:
this gives you the incremental time since last action, not last heartbeat. I'm not sure that's what you want.